dicompylercore/dose.py

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# dose.py
+"""Routines to access and modify DICOM RT Dose."""
+# Copyright (c) 2009-2016 Aditya Panchal
+# Copyright (c) 2019-2020 Dan Cutright
+# This file is part of dicompyler-core, released under a BSD license.
+#    See the file license.txt included with this distribution, also
+#    available at https://github.com/dicompyler/dicompyler-core/
+#
+# This code was adapted from dicom_dose_sum.py from DVH Analytics:
+#    https://github.com/cutright/DVH-Analytics/
+
+from copy import deepcopy
+import numpy as np
+from dicompylercore import dicomparser
+from pydicom.uid import generate_uid
+from pydicom.datadict import dictionary_VR, keyword_dict
+from dicompylercore.config import dicompyler_uid_prefix_rtdose, scipy_available
+from datetime import datetime
+from pydicom.sequence import Sequence
+from pydicom.dataset import Dataset
+from warnings import warn
+
+if scipy_available:
+    from scipy.ndimage import map_coordinates
+
+
+class DoseGrid:
+    """
+    Class to easily access commonly used attributes of a DICOM dose grid and
+    make modifications
+
+    Example: Add two dose grids
+        grid_1 = DoseGrid(dose_file_1)
+        grid_2 = DoseGrid(dose_file_2)
+        grid_sum = grid_1 + grid_2
+        grid_sum.save_dcm(some_file_path)
+
+    """
+
+    def __init__(self, rt_dose, order=1):
+        """ Initialization of a DoseGrid from a DICOM-RT Dose file or dataset.
+
+        Parameters
+        ----------
+        rt_dose : pydicom Dataset or filename
+            DICOM RT Dose used to determine the structure dose grid data.
+        order : int, optional
+            The order of the spline interpolation.
+            0: the nearest grid point, 1: trilinear, 2 to 5: spline
+        """
+
+        self.ds = dicomparser.DicomParser(rt_dose).ds
+        self.order = order
+
+        self.summation_type = None
+        self.sop_class_uid = self.ds.SOPClassUID
+        self.sop_instance_uid = self.ds.SOPInstanceUID
+        self.other_sop_class_uid = None
+        self.other_sop_instance_uid = None
+
+        if self.ds.Modality == "RTDOSE":
```

Comment:
Is this check and error necessary? I would assume if someone is using this class, they would be summing RT Dose.

Reply:
I would probably leave it if you think it will help. In that case, let's do a quick check for code coverage purposes. We could change the modality and test for failure.